bodhi/tests/client/test___init__.py

```diff
@@ -1696,6 +1696,158 @@ def test_info_with_errors(self, send_request):
+        Assert correct behavior with the --url flag.
+        """
+        runner = testing.CliRunner()
+
+        result = runner.invoke(client.list_critpath_packages,
+                               ['--url', 'http://localhost:6543', 'F27'])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(result.output,
+                         client_test_data.EXPECTED_CRITPATH_PACKAGES_OUTPUT)
+        bindings_client = send_request.mock_calls[0][1][0]
+        send_request.assert_called_once_with(bindings_client, 'releases/F27/critpath-packages',
+                                             verb='GET', auth=False)
+        self.assertEqual(bindings_client.base_url, 'http://localhost:6543/')
+
+    @mock.patch('bodhi.client.bindings.BodhiClient.csrf',
+                mock.MagicMock(return_value='a_csrf_token'))
+    @mock.patch('bodhi.client.bindings.BodhiClient.send_request',
+                return_value={'packages': []}, autospec=True)
+    def test_when_empty_result(self, send_request):
+        """
+        Assert correct behavior when there are no critical path packages.
+        """
+        runner = testing.CliRunner()
+
+        result = runner.invoke(client.list_critpath_packages,
+                               ['--url', 'http://localhost:6543', 'F27'])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(result.output,
+                         "No critical path packages for this release.\n")
+        bindings_client = send_request.mock_calls[0][1][0]
+        send_request.assert_called_once_with(bindings_client, 'releases/F27/critpath-packages',
+                                             verb='GET', auth=False)
+        self.assertEqual(bindings_client.base_url, 'http://localhost:6543/')
+
+    @mock.patch('bodhi.client.bindings.BodhiClient.csrf',
+                mock.MagicMock(return_value='a_csrf_token'))
+    @mock.patch('bodhi.client.bindings.BodhiClient.send_request',
+                return_value={"errors": [{"description": "an error was encountered... :("}]},
+                autospec=True)
+    def test_list_with_errors(self, send_request):
+        """
+        Assert errors are printed if returned back in the request
+        """
+        runner = testing.CliRunner()
+
+        result = runner.invoke(client.list_critpath_packages,
+                               ['--url', 'http://localhost:6543', 'F27'])
+
+        self.assertEqual(result.exit_code, 1)
+        self.assertEqual(result.output, ("ERROR: an error was encountered... :(\n"))
+
+
+class TestAddCritpathPackages(unittest.TestCase):
+    """
+    Test the add_critpath_packages() function.
+    """
+    @mock.patch('bodhi.client.bindings.BodhiClient.csrf',
+                mock.MagicMock(return_value='a_csrf_token'))
+    @mock.patch('bodhi.client.bindings.BodhiClient.send_request',
+                return_value=client_test_data.EXAMPLE_CRITPATH_PACKAGES_MUNCH, autospec=True)
+    def test_url_flag(self, send_request):
```

Comment:
It seems that the other tests set the ```--url``` flag anyway, so this test may be redundant? Same for the similarly named other tests.

Reply:
Is it a good idea to test normal execution? That test just checks whether, if normally executed, function returns expected output. It's name is not appropriate though - maybe ```test_list_critpath_packages```

Reply:
Yeah if that's the goal, let's change the name and keep it.

Reply:
Don't forget to change the name here and the description below.

Reply:
@bowlofeggs - I removed ```--url``` flag from the tests below it. So this one is just for testing that flag.